### PR TITLE
Main documentation

### PR DESCRIPTION
**Pull request by SMartorana**

### CODE REVIEW

**Comment by Sernior** [comment]
Why did you delete the thread_pool? I still use it in the main to do tests... it will be eventually deleted but not today.

File: include/boost/fairness/control_block_t.hpp [hunk 1]
```diff
@@ -19,32 +18,78 @@
 
 namespace boost::fairness{
```
Comment (Sernior) [inline include/boost/fairness/control_block_t.hpp:20]: Internals of the lib do not require doxygen comments.
If anything control_block definitions now need to be moved in the detail folder.

File: thread-pool/BS_thread_pool.hpp
Comment (Sernior) [inline thread-pool/BS_thread_pool.hpp:1]: Why was this deleted... it will be deleted in the future but I still use it in main.cpp to perform quick tests.

**Review by Sernior — CHANGES_REQUESTED** [review]

File: include/boost/fairness/control_block_t.hpp (hunk 1 above)
Comment (SMartorana) [reply include/boost/fairness/control_block_t.hpp:20]: ?

File: thread-pool/BS_thread_pool.hpp
Comment (SMartorana) [reply thread-pool/BS_thread_pool.hpp:1]: ?

**Comment by Sernior** [comment]
?

**Pull request closed without merging**